submitter/submission.py

```diff
@@ -61,7 +67,7 @@ def from_message(cls, message):
         except KeyError as e:
             raise errors.SubmitMessageMissingAttributeError(
                 message.message_id, e.args[0]
-            )
+            ) from e
 
         try:
```

Comment:
[Non-blocking] Commenting on this line but question is about the [exception](https://github.com/MITLibraries/dspace-submission-service/pull/42/files#diff-be0d35e1a31ee377fabcc80b1e5b32a9710de07344e3685a9b5225d3281b5da4R78): Curious what would result in a `TypeError`? I see that the `json.loads(message.body)` can result in a `JSONDecodeError` and attempting to index the `destination`, `collection_handle`, etc. can result in `KeyError`s. 🤔

Reply:
It's a good question.  My assumption is that it's an edge case handling if the decoded JSON does not result in a dictionary.

Example:

```python
import json
body = json.loads("null")  # technically valid JSON; could be a single integer, string, etc., anything but an object
destination = body['SubmissionSystem']
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Cell In[9], line 1
----> 1 destination = body['SubmissionSystem']

TypeError: 'NoneType' object is not subscriptable
```